main.tf

```diff
@@ -0,0 +1,214 @@
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 5.16"
+    }
+  }
+
+  required_version = ">= 1.5.7"
+
+  backend "s3" {
+    bucket               = "ccao-terraform-state-us-east-1"
+    key                  = "terraform.tfstate"
+    region               = "us-east-1"
+    workspace_key_prefix = "model-sales-val/workspaces"
+  }
+}
+
+provider "aws" {
+  region = "us-east-1"
+}
+
+locals {
+  s3_prefix                = "scripts/sales-val"
+  s3_bucket_data_warehouse = terraform.workspace == "prod" ? "ccao-data-warehouse-us-east-1" : aws_s3_bucket.data_warehouse[0].id
+  s3_bucket_glue_assets    = terraform.workspace == "prod" ? "ccao-glue-assets-us-east-1" : aws_s3_bucket.glue_assets[0].id
+  glue_job_name            = "sales_val_flagging${terraform.workspace == "prod" ? "" : "_${terraform.workspace}"}"
+  glue_crawler_name        = "ccao-data-warehouse-sale-crawler${terraform.workspace == "prod" ? "" : "-${terraform.workspace}"}"
+  # Athena databases cannot have hyphens, so replace them with underscores
+  # (Note that this is not always true -- notably, dbt-athena is able to
+  # create Athena tables with hyphens -- but it's a rule that Terraform
+  # enforces, so we follow it here)
+  athena_database_name = terraform.workspace == "prod" ? "sale" : "ci_model_sales_val_${replace(terraform.workspace, "-", "_")}_sale"
```

Comment:
New as of last review cycle, this database helps keep staging data isolated on branches. The mismatch with our dbt staging database pattern is a bit of a bummer 😕